### PR TITLE
Doxygen updates 2

### PR DESCRIPTION
**Pull request by dovholuknf**
Updates to provide API doc for the C sdk. These changes are available here https://dovholuknf.github.io/ziti-doc/api/clang/api/index.html for (hopefully) easier preview

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -12,6 +12,8 @@ message("   branch : ${GIT_BRANCH}")
 message("     hash : ${GIT_COMMIT_HASH}")
 message("     date : ${BUILD_DATE}")
 
+set(ZITI_HEADER_FILES ${PROJECT_SOURCE_DIR}/includes/nf/errors.h ${PROJECT_SOURCE_DIR}/includes/nf/ziti.h)
```
Comment (ekoby) [inline CMakeLists.txt]: this is no longer used, right?

Comment (dovholuknf) [reply CMakeLists.txt]: it's referenced in library/CMakeLists.txt but not in the example programs

Comment (ekoby) [reply CMakeLists.txt]: it should be moved to the one in library cmake then

Comment (dovholuknf) [reply CMakeLists.txt]: moved

**Pull request merged**